qt/python/mantidqt/utils/asynchronous.py

```diff
@@ -90,6 +90,26 @@ def abort(self):
         time.sleep(0.1)
 
 
+class Receiver(object):
```

Comment:
LGTM, just one last thing I should have picked up, is this an implementation detail? If so can we prepend an `_` so it's `_Receiver` and people don't try to use it

Apart from this I'm ready to approve